owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/SettingsActivity.kt

```diff
@@ -58,12 +63,40 @@ class SettingsActivity : AppCompatActivity() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             android.R.id.home -> {
-                intent = Intent(this, FileDisplayActivity::class.java).apply {
-                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                if (supportFragmentManager.findFragmentByTag("logs") != null) {
+                    supportFragmentManager.popBackStack()
+                    supportActionBar?.setTitle(R.string.actionbar_settings)
+                } else {
+                    intent = Intent(this, FileDisplayActivity::class.java).apply {
+                        addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                    }
+                    startActivity(intent)
                 }
-                startActivity(intent)
             }
         }
         return super.onOptionsItemSelected(item)
     }
+
+    override fun onPreferenceStartFragment(caller: PreferenceFragmentCompat, pref: Preference): Boolean {
+        val fragment = supportFragmentManager.fragmentFactory.instantiate(
+            classLoader,
+            pref.fragment)
+        var tag: String? = null
+        var titleId: Int? = null
+        if (pref.key.equals(SettingsFragment.PREFERENCE_LOGS_SUBSECTION)) {
+            tag = "logs"
+            titleId = R.string.actionbar_logger
+        }
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.settings_container, fragment, tag)
+            .addToBackStack(null)
+            .commit()
+        supportActionBar?.setTitle(titleId!!)
+        return true
+    }
```

Comment:
I understand it, but from my point of view, it is not a good solution. Tags are not needed at all to handle this, and the action bar title should be handled in a different way. We could add an `addOnBackStackChangedListener` and check there the fragment and update the action bar depending on it.

```suggestion
    override fun onPreferenceStartFragment(caller: PreferenceFragmentCompat, pref: Preference): Boolean {
        val fragment = supportFragmentManager.fragmentFactory.instantiate(classLoader, pref.fragment)

        supportFragmentManager.beginTransaction()
            .replace(R.id.settings_container, fragment)
            .addToBackStack(null)
            .commit()

        return true
    }
```